### PR TITLE
DevTools: Add root and renderer version to inspected props panel

### PR DESCRIPTION
**Pull request by bvaughn**
Adds the root API and renderer version to the bottom of the "rendered by" list. (If this info is missing, we degrade gracefully.)

# Screenshots

Screen Shot 2020-05-20 at 10 04 22 AM

Screen Shot 2020-05-20 at 10 04 09 AM

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/Components/Badge.js [hunk 1]
```diff
@@ -26,12 +26,12 @@ export default function Badge({
   type,
   children,
 }: Props) {
-  let totalBadgeCount = 0;
-
-  if (hocDisplayNames !== null) {
-    totalBadgeCount += hocDisplayNames.length;
+  if (hocDisplayNames === null) {
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/Components/Badge.js:29]: This fixes a regression from #18802 I think.

File: packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js [hunk 2]
```diff
@@ -374,6 +376,15 @@ function InspectedElementView({
     };
   }
 
+  const rendererLabel =
+    rendererPackageName !== null && rendererVersion !== null
+      ? `${rendererPackageName}@${rendererVersion}`
+      : null;
+  const showRenderedBy =
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js:383]: We'll always show this section now, except for v15.

File: packages/react-reconciler/src/ReactFiberRoot.new.js [hunk 3]
```diff
@@ -60,6 +61,20 @@ function FiberRootNode(containerInfo, tag, hydrate) {
   if (enableSuspenseCallback) {
     this.hydrationCallbacks = null;
   }
+
+  if (__DEV__) {
+    switch (tag) {
+      case BlockingRoot:
+        this._debugRootType = 'createBlockingRoot()';
```
Comment (bvaughn) [inline packages/react-reconciler/src/ReactFiberRoot.new.js:68]: This minimizes the coupling between React and DevTools.

File: packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js [hunk 4]
```diff
@@ -415,19 +426,27 @@ function InspectedElementView({
...
-                key={owner.id}
-                displayName={owner.displayName || 'Anonymous'}
-                hocDisplayNames={owner.hocDisplayNames}
-                id={owner.id}
-                isInStore={store.containsElement(owner.id)}
-                type={owner.type}
-              />
-            ))}
+            {owners !== null &&
+              owners.length > 0 &&
```
Comment (lunaruan) [inline packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js:433]: Are you intentionally checing this condition twice (also in `showRenderedBy`)

Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js:433]: Yes for Flow. I guess I could technically get rid of the second `owners.length` check if I moved `owners !== null && owners.length > 0` above into a second variable but it seemed kind of not worth doing.

Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js]: I'll just add the Flow casting to get rid of the second check though :smile: I don't mind.

**Pull request merged**